sway/handlers.c

```diff
@@ -387,6 +388,24 @@ static bool handle_key(wlc_handle view, uint32_t time, const struct wlc_modifier
 			}
 		}
 	}
+
+	// handle bar modifiers pressed/released
+	struct bar_config *bar;
+	for (i = 0; i < config->bars->length; ++i) {
+		bar = config->bars->items[i];
+		if (strcmp(bar->mode, "hide") == 0 && strcmp(bar->hidden_state, "hide") == 0) {
+			switch (modifier_state_changed(modifiers->mods, bar->modifier)) {
+			case MOD_STATE_PRESSED:
+				ipc_event_modifier(bar->modifier, "pressed");
+				break;
+			case MOD_STATE_RELEASED:
+				ipc_event_modifier(bar->modifier, "released");
+				break;
+			}
+		}
+	}
+	// update modifiers state
+	modifiers_state_update(modifiers->mods);
```

Comment:
It looks like this code emits a bar modifier *for* every bar to every bar, instead of *to* every bar. In other words, press and release the mod key with three bars running and 6 events will be sent to *each of them*.